### PR TITLE
Feature/declaration

### PR DESCRIPTION
**Pull request by LaurentMag**
# Description
Front : 
* generic : page layout, status card,
* declaration page
* declaration model
* declaration retropood and retrofit for api connexion

Back :
* create  api service to handle declaration creation ( router - repository - service - controller )
* register declaration in DB

# Linked Issues
#4

# Changes
declaration front end feat

# Screenshots
N/A

# Tests
How has it been tested ?

- [X] Manually
- [ ] Automated tests
- [ ] QA
- [ ] Other

# Additional information
N/A

### CODE REVIEW

File: yaki_mobile/lib/presentation/ui/declaration/views/declaration_body.dart [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+
+  ref.read(declarationProvider.notifier).create(declaration);
+}
+
+// ConsumerWidget : stateless widget listening to providers
+class DeclarationBody extends ConsumerWidget {
+  const DeclarationBody({Key? key}) : super(key: key);
+
+  static const spacingMultiplier = 0.07;
+  static const runSpacingMultiplier = 0.12;
```
Comment (PiotrFLEURY) [inline yaki_mobile/lib/presentation/ui/declaration/views/declaration_body.dart:36]: magic numbers, remove it

File: yaki_mobile/lib/presentation/ui/declaration/views/status_card.dart [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+  final VoidCallback getClick;
+
+  const StatusCard({
+    super.key,
+    required this.statusName,
+    required this.statusPicto,
+    required this.getClick,
+  });
+
+  static const sizeMultiplier = 0.37;
```
Comment (PiotrFLEURY) [inline yaki_mobile/lib/presentation/ui/declaration/views/status_card.dart:16]: magic number remove it

File: yaki_mobile/lib/presentation/ui/shared/pages_layout.dart [hunk 3]
```diff
@@ -0,0 +1,31 @@
+import 'package:flutter/material.dart';
+import 'views/header.dart';
```
Comment (PiotrFLEURY) [inline yaki_mobile/lib/presentation/ui/shared/pages_layout.dart:2]: use package imports

File: yaki_mobile/lib/temp_home.dart [hunk 4]
```diff
@@ -0,0 +1,54 @@
+import 'package:flutter/material.dart';
+import 'package:go_router/go_router.dart';
+
+class TempHome extends StatelessWidget {
```
Comment (PiotrFLEURY) [inline yaki_mobile/lib/temp_home.dart:4]: will be replaced by the splash screen in future sprint

File: yaki_mobile/lib/temp_home.dart [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+      ),
+      body: Padding(
+        padding: const EdgeInsets.all(40.0),
+        child: Column(
+          crossAxisAlignment: CrossAxisAlignment.stretch,
+          mainAxisAlignment: MainAxisAlignment.spaceEvenly,
+          children: const [
+            ButtonRoute(
+              btnText: 'Authetification page',
+              route: '/authentification',
```
Comment (PiotrFLEURY) [inline yaki_mobile/lib/temp_home.dart:22]: `authentication`

File: yaki_mobile/pubspec.yaml [hunk 6]
```diff
@@ -93,6 +97,7 @@ dev_dependencies:
   retrofit_generator: ">=5.0.0 <6.0.0"
   build_runner: ">=2.3.0 <4.0.0"
   json_serializable: ^6.6.1
+  dio: ^5.0.1
```
Comment (PiotrFLEURY) [inline yaki_mobile/pubspec.yaml:100]: should be in the dependencies not in dev_dependencies

**Review by PiotrFLEURY — CHANGES_REQUESTED** [review]

File: yaki_mobile/lib/data/models/declaration_model.dart [hunk 7]
```diff
@@ -0,0 +1,20 @@
+import 'package:json_annotation/json_annotation.dart';
+
+part 'declaration_model.g.dart';
+
+@JsonSerializable()
+class DeclarationModel {
+  DateTime declaration_date;
+  int declaration_team_mate_id;
+  String declaration_status;
```
[the remark this answers is not in the document]
Comment (LaurentMag) [reply yaki_mobile/lib/data/models/declaration_model.dart:9]: its to match database attribute

File: yaki_mobile/lib/data/sources/declaration_api.dart [hunk 8]
```diff
@@ -0,0 +1,15 @@
+import 'package:dio/dio.dart';
+import 'package:retrofit/retrofit.dart';
+import 'package:yaki/data/models/declaration_model.dart';
+
+part 'declaration_api.g.dart';
+
+const baseUrl = 'http://192.168.1.116:3000/';
```
Comment (PiotrFLEURY) [inline yaki_mobile/lib/data/sources/declaration_api.dart:7]: still not a good solution

**Review by PiotrFLEURY — CHANGES_REQUESTED** [review]

**Review by PiotrFLEURY — APPROVED** [review]

**Review by MaximeHavez — APPROVED** [review]

**Pull request merged**